src/libraries/Microsoft.PowerFx.Connectors/PowerPlatformConnectorClient.cs

```diff
@@ -18,7 +20,23 @@ public class PowerPlatformConnectorClient : HttpClient
         /// <summary>
         /// For telemetry - assembly version stamp. 
         /// </summary>
-        public static string Version { get; } = typeof(PowerPlatformConnectorClient).Assembly.GetName().Version.ToString();
+        public static string Version
+        {
+            get
+            {
+                if (!string.IsNullOrEmpty(_version))
+                {
+                    return _version;
+                }
+
+                // InformationVersion example: "0.2.6-preview.20230615-1002+048250dc4404bdb749f992cd0869f8821c79e3fe"
+                _version = typeof(PowerPlatformConnectorClient).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion.Split('+')[0];
+
```

Comment:
can you set this in constructor and set the _version variable to readonly?

Reply:
Version is static and our constructors aren't
Not that we couldn't have one static constructor but I don't see the value here

Reply:
Fixed